extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -113,6 +158,19 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 			if (this._openAsUntitled) {
 				this.openNotebookAsUntitled(resource);
 			} else {
+				// let us keep a list of already visited notebooks so that we do not trust them again, potentially
+				// overriding user changes
+				let normalizedResource = path.normalize(resource);
+				let self = this;
```

Comment:
You shouldn't need to use self here if you use a lambda instead of function declaration 

`(document: azdata.nb.NotebookDocument) => { ... }`